tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNetCore/AspNetCoreBlockMiddlewareIntegrationEnd.cs

```diff
@@ -0,0 +1,63 @@
+// <copyright file="AspNetCoreBlockMiddlewareIntegrationEnd.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if !NETFRAMEWORK
+using System;
+using System.ComponentModel;
+using System.Threading.Tasks;
+using Datadog.Trace.AppSec;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.Configuration;
+using Microsoft.AspNetCore.Http;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNetCore
+{
+    /// <summary>
+    /// The ASP.NET Core middleware integration.
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = AssemblyName,
+        TypeName = ApplicationBuilder,
+        MethodName = "Build",
+        ReturnTypeName = "Microsoft.AspNetCore.Http.RequestDelegate",
+        MinimumVersion = Major3,
+        MaximumVersion = Major6,
```

Comment:
Will need bumping to .NET 7 after I merge! 🙂

Reply:
done in 53bb7e2c88e7f1eb4aea8f5d0053016ec0b0a6fc